### DISCUSSION TITLE
is this violin plot is correct?

### DISCUSSION THREAD

**Discussion by HamzaAli195** [AammarTufail/python_ka_chilla2023#60] [Q&A] [answered]
can anyone see my plot to verify is it true or false?
<img width="507" alt="violinplot v" src="https://user-images.githubusercontent.com/121845773/212253206-e7d7d796-9a98-498f-91e3-d92d16f54ec5.PNG">

**Comment by usamabajwa86** [comment, marked as answer]
chart is correct but its not readable, as there are somany entries in x-axes, you can replace x axes value by species and add two variables in y-axes

**Reply by HamzaAli195** [reply]
I  know we use only categoric variables to use colors in the plots, to make them more readable.
But still I have some confusion about it, can you give me
I am sending you the PDF file, can you check it and correct my mistakes?

[day_11_iris_dataset_plots_after_correction.pdf](https://github.com/AammarTufail/python_ka_chilla2023/files/10409137/day_11_iris_dataset_plots_after_correction.pdf)
 some tip how to make charts more readable?

**Reply by usamabajwa86** [reply]
you can't use violinplot or boxenplot for variable where x axis values are high, for this you cane use line plot or bar plot

**Reply by Muhammad-Ali-Butt** [reply]
> I know we use only categoric variables to use colors in the plots, to make them more readable. But still I have some confusion about it, can you give me I am sending you the PDF file, can you check it and correct my mistakes?
> 
> [day_11_iris_dataset_plots_after_correction.pdf](https://github.com/AammarTufail/python_ka_chilla2023/files/10409137/day_11_iris_dataset_plots_after_correction.pdf) some tip how to make charts more readable?

add plt.figure(figsize=(16, 6))  16,6  means 16 inch width and 6 inch height. Then next add code for violin plot **(violin=sns.violinplot(data=phool,palette='magma',x='petal_width',y='petal_length',hue='species')
)** and also this is optional you should try with first adjusting figsize as mentioned before and then if you want you can adjust the rotation plus fontsize of the xticks using violin.set_xticklabels(violin.get_xticklabels(),rotation=45, ha="right", fontsize=7) btw ha means horizontal alignment of the xticks to right and if im not mistaken its value by default is to right as well, you can try that by including or excluding the ha argument.

im giving just a rough idea , you can further customize the aesthetics  btw do let me know if this helped or didnt change anything

**Reply by Muhammad-Ali-Butt** [reply]
> [day_11_iris_dataset_plots_after_correction.pdf](https://github.com/AammarTufail/python_ka_chilla2023/files/10409137/day_11_iris_dataset_plots_after_correction.pdf) some tip how to make charts more readable?

bro i skimmed through your pdf and it seems like you have used rotation 90 on every barplot which seems very odd as it kind of ruins the overall plot aesthetic so what i would recommend is the the thing i said in the previous comment i.e. use wider figsize and if you still want to change ticks look then go for plt.xticks() etc

**Reply by HamzaAli195** [reply]
thanks to "Usama bhai" and "Muhammad Ali bhai ".Your suggestions helped me .
Jazak Allah

**Reply by Muhammad-Ali-Butt** [reply]
> thanks to "Usama bhai" and "Muhammad Ali bhai ".Your suggestions helped me . Jazak Allah

Wa iyyakum and yw bhai glad it helped  :)
